pkg/insight/filestore.go

```diff
@@ -68,6 +72,36 @@ func (s *Store) PutChunk(ctx context.Context, chunk Chunk) error {
 	return s.filestore.PutObject(ctx, path, data)
 }
 
+func LoadChunksFromCache(cache cache.Cache, projectID, appID string, kind model.InsightMetricsKind, step model.InsightStep, from time.Time, count int) (Chunks, error) {
+	paths := determineFilePaths(projectID, appID, kind, step, from, count)
+	chunks := make([]Chunk, 0, len(paths))
+	for i, p := range paths {
+		c, err := cache.Get(p)
+		if err != nil {
+			return nil, err
+		}
+
+		chunk, ok := c.(Chunk)
+		if !ok {
+			return nil, errors.New("malformed chunk data in cache")
+		}
+		chunks[i] = chunk
```

Comment:
because the size of chunks is zero so let's keep it as the previous
`chunks = append(chunks, chunk)`

Reply:
Oops, thanks 🙏